src/components/fullscreen/fullscreen.js

```diff
@@ -12,7 +12,8 @@ import {default as Icon, IconType} from '../icon';
  * @returns {Object} - mapped state to this component
  */
 const mapStateToProps = state => ({
-  fullscreen: state.engine.fullscreen
+  fullscreen: state.engine.fullscreen,
+  config: state.config
```

Comment:
attaching state to props means also that component will update on each state change.
As you only want the targetId then target only that attribute
```suggestion
  targetId: state.config.targetId
```